### PR TITLE
Fixes VSTS Bug 801652: New Editor: broken history of references navigation

### PR DESCRIPTION
**Pull request by mkrueger**

### CODE REVIEW

**Comment by mkrueger** [comment]
@monojenkins backport master

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Navigation/DocumentNavigationPoint.cs [hunk 1]
```diff
@@ -53,22 +53,24 @@ public DocumentNavigationPoint (Document doc)
...
 			this.fileName = fileName;
 		}
-		
-		public override void Dispose ()
+
+		protected void SetDocument (Document doc)
 		{
-			if (doc != null) {
+			if (this.document != null)
 				doc.Closed -= HandleClosed;
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Navigation/DocumentNavigationPoint.cs]:
```suggestion
				this.document.Closed -= HandleClosed;
```

**Comment by mkrueger** [comment]
@monojenkins backport master

**Comment by mkrueger** [comment]
Navigation was never perfect - is that a regression or part of that bug ?

Navigation needs to be done with tracking points for tracking the changes atm it works with fixed coordinates. Which is no change to a former version. 

Therefore I would suggest opening new issues for these items.

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Navigation/TextFileNavigationPoint.cs [hunk 2]
```diff
@@ -111,32 +111,24 @@ protected override async Task<Document> DoShow ()
...
 		}
 
-		protected void JumpToCurrentLocation (TextEditor editor)
+		protected void JumpToCurrentLocation (ITextView textView)
 		{
-			if (version != null && version.BelongsToSameDocumentAs (editor.Version)) {
-				var currentOffset = version.MoveOffsetTo (editor.Version, offset);
-				var loc = editor.OffsetToLocation (currentOffset);
-				editor.SetCaretLocation (loc);
+			if (version != null && textView == this.textView && offset.HasValue) {
```
Comment (DavidKarlas) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Navigation/TextFileNavigationPoint.cs]: I think this method can become:
```
if(textView==this.textView && offset.HasValue){
textView.TryMoveCaretToAndEnsureVisible (offset.TranslateTo(textView.TextSnapshot, PointTrackingMode.Positive));
}
```

**Comment by mkrueger** [comment]
@monojenkins backport master

**Comment by sevoku** [comment]
@monojenkins backport release-8.1

**Comment by sevoku** [comment]
Superseded by #7805, Closing

**Pull request closed without merging**